imscp.py

```diff
@@ -0,0 +1,112 @@
+        zip_file.extractall(extract_path)
+        return extract_from_dir(extract_path, license)
+
+
+def extract_from_dir(ims_dir, license):
+    """Return a list of content nodes from an IMSCP directory."""
+    print('Parsing imsmanifest.xml in %s' % ims_dir)
+    manifest_path = os.path.join(ims_dir, 'imsmanifest.xml')
+    manifest_root = etree.parse(manifest_path).getroot()
+    nsmap = manifest_root.nsmap
+
+    print('Extracting tree structure ...')
+    print()
+
+    # NOTE: Can there be multiple organizations?
+    organization = manifest_root.find('organizations/organization', nsmap)
+    organization_items = walk_items(organization)
+
+    resources_elem = manifest_root.find('resources', nsmap)
+    resources_dict = dict((r.get('identifier'), r) for r in resources_elem)
+
+    collect_resources(license, organization_items, resources_dict, ims_dir)
+
+    return organization_items
+
+
+def walk_items(root):
+    items_list = []
+
+    for item in root.findall('item', root.nsmap):
+        item_dict = dict(item.items())
+
+        title_elem = item.find('title', item.nsmap)
+        if title_elem is not None:
+            item_dict['title'] = title_elem.text
+
+        item_dict['children'] = walk_items(item)
+        items_list.append(item_dict)
+
+    return items_list
+
+
+def collect_resources(license, items_list, resources_dict, ims_dir):
+    for item in items_list:
+        resource_elem = resources_dict[item['identifierref']]
+        if item['children']:
+            collect_resources(license, item['children'], resources_dict, ims_dir)
+        else:
+            if resource_elem.get('type') == 'webcontent':
+                item['node'] = package_into_html5_app_node(
+                        license, item, resource_elem, resources_dict, ims_dir)
+
+
+def package_into_html5_app_node(license, item, resource_elem, resources_dict, ims_dir):
+    nsmap = resource_elem.nsmap
+    with tempfile.TemporaryDirectory() as destination:
+        # Copy index file into our temporary directory and call it index.html
+        index_path = os.path.join(ims_dir, resource_elem.get('href'))
+        index_copy_path = os.path.join(destination, 'index.html')
+        shutil.copyfile(index_path, index_copy_path)
+
+        def copy_files(resource_elem):
+            for file_elem in resource_elem.findall('file', nsmap):
+                file_path = os.path.join(ims_dir, file_elem.get('href'))
+                shutil.copy(file_path, destination)
+
+            for dependency_elem in resource_elem.findall('dependency', nsmap):
+                dep_res_elem = resources_dict[dependency_elem.get('identifierref')]
+                copy_files(dep_res_elem)
+
+        # Copy files and recursively copy files in dependency elements into temp
+        # dir
+        copy_files(resource_elem)
+
+        #preview_in_browser(destination)
+
+        zip_path = create_predictable_zip(destination)
+        return nodes.HTML5AppNode(
```

Comment:
Actually, thinking more about this, let's move all the HTML5 zip creation work to phase 2, when we add ricecooker packaging support. We will need the zips regardless of how we package, but I think it might be best to keep the parsing code focused solely on parsing, and do any transformations as a later step.

For now, let's just return the path to the actual file(s) so that it's up to the consumer of that data to decide what to do with it. Maybe go ahead and create a `ricecooker_utils.py` or something and put this function in there for now.